### PR TITLE
Issues/veen 51 npe on missing tag reference

### PR DESCRIPTION
**Pull request by gijskant**

### CODE REVIEW

**Comment by gijskant** [comment]
@forus @cataphract This pull request is still open. It contains fixes for fetching tags associated with tag types for the VEEN project.

File: transmart-core-db-tests/test/integration/org/transmartproject/db/ontology/OntologyTermTagsResourceServiceTests.groovy [hunk 1]
```diff
@@ -118,13 +118,25 @@ class OntologyTermTagsResourceServiceTests {
                     hasProperty('key', equalTo(key)),
```
Comment (forus) [inline transmart-core-db-tests/test/integration/org/transmartproject/db/ontology/OntologyTermTagsResourceServiceTests.groovy:118]: I am confused: `key` variable has `String` type. But map signature suggests that the map contains an object of the `OntologyTerm` class.

Comment (gijskant) [inline transmart-core-db-tests/test/integration/org/transmartproject/db/ontology/OntologyTermTagsResourceServiceTests.groovy:118]: Hmm, interesting. btw, I didn't write the code, only moved it. But it appears as if the `equals` method of the implementation (probably `AbstractI2b2Metadata`, using `@EqualsAndHashCode`) does accept the `String` argument, right? Maybe better would be something like `hasProperty('fullName', equalTo(...))`...

Comment (forus) [inline transmart-core-db-tests/test/integration/org/transmartproject/db/ontology/OntologyTermTagsResourceServiceTests.groovy:118]: @gijskant IMHO that's not stopper for this PR to be merged. As soon as it works.

**Pull request merged**